File: src/Loaders/JavaPluginLoader/java-src/fermanext/system/PluginExecutionParams.java

package fermanext.system;

import java.util.HashMap;
import java.util.Map;

import fermanext.system.Variant;

public final class PluginExecutionParams
{
    /** Adds param. */
    public void addParam ( String name, Variant value )
    { params.put( name, value );  }

    /** Removes params. */
    public void removeParam ( String name )
    { params.remove( name ); }

    /** Returns whole map of params */
    public Map<String, Variant> getParams ()
    { return params; }

//Private members:
    private HashMap<String, Variant> params = new HashMap<String, Variant>();
}

File: src/Common/Plugin/PluginExecutionParams.cpp

#include "PluginExecutionParams.h"

/*****************************************************************************/

void PluginExecutionParams::addParam ( const QString& name, 
                                       const QVariant& value )
{ params.insert( name, value ); }

void PluginExecutionParams::removeParam ( const QString& name )
{ params.remove(name); }

const QHash<QString, QVariant>& PluginExecutionParams::getParams () const
{ return params; }

/*****************************************************************************/
